code/game/machinery/status_display.dm

```diff
@@ -93,57 +92,29 @@
 
 /// Immediately change the display to the given two lines.
 /obj/machinery/status_display/proc/update_display(line1, line2)
+	if( \
+		(message1_overlay && message1_overlay.message == line1) && \
+		(message2_overlay && message2_overlay.message == line2))
+		return
+
+	remove_display()
+
 	line1 = uppertext(line1)
 	line2 = uppertext(line2)
-	var/new_text = {"<div style="font-size:[FONT_SIZE];color:[FONT_COLOR];font:'[FONT_STYLE]';text-align:center;" valign="top">[line1]<br>[line2]</div>"}
-	if(maptext != new_text)
-		maptext = new_text
-
-/// Prepare the display to marquee the given two lines.
-///
-/// Call with no arguments to disable.
-/obj/machinery/status_display/proc/set_message(m1, m2)
-	if(m1)
-		index1 = (length_char(m1) > CHARS_PER_LINE)
-		message1 = m1
-	else
-		message1 = ""
-		index1 = 0
 
-	if(m2)
-		index2 = (length_char(m2) > CHARS_PER_LINE)
-		message2 = m2
-	else
-		message2 = ""
-		index2 = 0
+	message1_overlay = new(LINE1_Y, line1)
+	vis_contents += message1_overlay
 
-// Timed process - performs default marquee action if so needed.
+	message2_overlay = new(LINE2_Y, line2)
+	vis_contents += message2_overlay
```

Comment:
Why vis_contents and not overlays?

Reply:
Overlays are baked as a static image-esque appearance object, including map text. In order for the animation to run on the client, it has to be a vis_content.